src/main/java/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,77 @@
+package bookstore.controller;
+
+import bookstore.dto.cart.AddToCartDto;
+import bookstore.dto.cart.CartResponseDto;
+import bookstore.dto.cartitem.CartItemResponseDto;
+import bookstore.model.User;
+import bookstore.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/cart")
+@RequiredArgsConstructor
+@Tag(name = "Shopping Cart Controller", description = "Endpoints for managing the shopping cart")
+public class ShoppingCartController {
+    private final ShoppingCartService service;
+
+    @Operation(summary = "Get the user's shopping cart")
+    @GetMapping
+    public CartResponseDto get(Authentication authentication) {
+        return service.getShoppingCartByUser((User) authentication.getPrincipal());
+    }
+
+    @Operation(summary = "Add an item to the shopping cart")
+    @PostMapping()
+    @ApiResponse(responseCode = "204", description = "Item added to the cart")
+    public void addCartItemTo(Authentication authentication,
+            @RequestBody @Parameter(description = "Item to add to the cart")
+            @Valid AddToCartDto addToCartDto) {
+        service.addItemToCart(addToCartDto, (User) authentication.getPrincipal());
+    }
+
+    @Operation(summary = "Update a shopping cart item by ID")
+    @PutMapping("/cart-items/{id}")
```

Comment:
![image](https://github.com/DmytroFedoryshyn/bookstore/assets/20326858/23ccf159-e983-4993-aa00-e3d8017aa064)

Reply:
I have "api/cart" part in my @RequestMapping annotation placed on top of controller